optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/phase/event/TestdataStepValueListener.java

```diff
@@ -0,0 +1,37 @@
+package org.optaplanner.core.impl.testdata.phase.event;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.optaplanner.core.impl.phase.event.PhaseLifecycleListenerAdapter;
+import org.optaplanner.core.impl.phase.scope.AbstractStepScope;
+import org.optaplanner.core.impl.solver.scope.DefaultSolverScope;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataComparableEntity;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataComparableSolution;
+
+public class TestdataStepValueListener extends PhaseLifecycleListenerAdapter<TestdataComparableSolution> {
```

Comment:
Looks like this is a solution state recorder.
But while listening to steps, you can also have a step (= winning move) recorder.
That only has the delta's.

Reply:
The way it is implemented now seems to me being similarly complex as having the winning move recorder with deltas, while being a bit more comprehensible to a human and achieving the same outcome than having the deltas instead, or am I missing something?